src/components/AddBookForm.jsx

```diff
@@ -1,16 +1,34 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+import { addBooks } from '../Redux/Books/Books';
 
 function AddBookForm() {
+  const dispatch = useDispatch();
+  const submitHandler = (event) => {
+    event.preventDefault();
+    const title = event.target.title.value;
+    const author = event.target.author.value;
+    const id = Math.floor(Math.random() * 1000);
+    const book = {
+      id,
+      title,
+      author,
+    };
+    dispatch(addBooks(book));
+    event.target.title.value = '';
+    event.target.author.value = '';
+  };
+
   return (
-    <form>
+    <form onSubmit={submitHandler}>
       <label htmlFor="title">
         Title
-        <input type="text" name="title" id="title" />
+        <input type="text" name="title" id="title" required />
       </label>
 
       <label htmlFor="author">
         Author
-        <input type="text" name="author" id="author" />
+        <input type="text" name="author" id="author" required />
       </label>
```

Comment:
- [ ] Your implementation seems to work fine here but you could make it more efficient that you pass an onChange prop on the input fields to update the state values instead of directly getting the input values as it is done in a vanilla JavaScript project. With the onChange prop, on every change on the input fields, will update their values and that's the native and efficient way you could implement that.

```javascript
  <input onChange={(e) => setTitle(e.target.value)} type="text" name="title" id="title" required />
```

Reply:
Hey @AmaduKamara, I understand that, but then the user has to click on submit which will also do the same thing, so there is no actual use of this, and this should be an optional requirement.